Mukundi's Real-world Scenario

m-mukundi opened the pull request:
Hey team. Please review my work and leave me feedback.

Here is a screenshot of the results.
<img width="960" alt="image" src="https://github.com/Armstrong-Kim/Tana-Fellowship-10/assets/153626150/1d7ed323-61f4-4a74-9cd9-a57a954a0120">


[Here](https://drive.google.com/file/d/1e0ceajDuc64NDnS8H92vU_xSCHOLiewz/view?usp=drive_link) is a link to my video

m-mukundi committed:
Added Contact us page positive automation test

sharon-odhiambo reviewed (approved):
Hey @m-mukundi, great job with this assignment. You have written all the tests correctly and each is running as it should. I like that you have used Scenario Outline to help you avoid repetition in your code. This shows a great level of understanding in utilizing the combination of Cypress + Cucumber + Gherkin in automating your tests. I also like that your video shows a brief explanation of how the website should work and goes ahead to run all the tests for the Contact-us and Login. It's always good to show that your tests are working as expected. 

A little tip for improvement is to give a brief summary in your Pull Request description on what your changes cover. Additionally, let's see how to improve the video tomorrow from Eric's tool. 
Great work today! Keep it up!

m-mukundi committed:
Added HTML, JSON and Multiple Reports

m-mukundi commented:
I have added the code for report creation in this branch. I added the html, JSON and multiple-html report. 

Here is a screenshot of the html report:
![HTML report](https://github.com/Armstrong-Kim/Tana-Fellowship-10/assets/153626150/c3c6013b-f78a-4a2a-9723-9de705cbc704)

Here is a screenshot of the json report:
<img width="960" alt="json report" src="https://github.com/Armstrong-Kim/Tana-Fellowship-10/assets/153626150/5db61f41-b02e-4a6c-8d5e-c736269482dc">

Here is a screenshot of the multiple-html-report:
![Multiple Cucumber HTML Reporter](https://github.com/Armstrong-Kim/Tana-Fellowship-10/assets/153626150/6d36549f-a1ff-440f-98f8-ca564ff0ff6b)

And [here](https://drive.google.com/file/d/18u9EgIJRPmQNTg7gTI5MSNBSw3IdzPPr/view?usp=sharing) is a video of everything.

<video controls>
  <source src="https://drive.google.com/file/d/18u9EgIJRPmQNTg7gTI5MSNBSw3IdzPPr/view?usp=sharing" type="video/mp4">
Your browser does not support the video tag.
</video>

Beryl01 commented:
Hi @m-mukundi I like that your reports are clear and well-organized, making it easy to understand the test results. The use of Cucumber's Gherkin syntax ensures that scenarios are written in a readable format. Your reports accurately reflect the outcome of your test scenarios and you have attached screenshots. Each step is documented. The video presentation of your reports is well done for conveying the report information effectively to your teammates.

Keep up the good job.

m-mukundi committed:
Concluded Cypress+Cucumber Tutorial and added POM sub-pages

m-mukundi commented:
I have finally concluded the Cypress and Cucumber tutorials, and have been practicing the Page Object Model (POM) design pattern. Throughout this process, I've implemented various test scenarios for the contact us page and login page functionalities.

Here's a screenshot of the contact us page tests:
<img width="960" alt="image" src="https://github.com/Armstrong-Kim/Tana-Fellowship-10/assets/153626150/20f8b020-18b0-4c98-81b0-1f0e62560c84">

And here's a screenshot of the login page tests:
<img width="960" alt="image" src="https://github.com/Armstrong-Kim/Tana-Fellowship-10/assets/153626150/63f7142b-d309-4b2b-b2c9-97dc966b34e1">

To provide a comprehensive overview, I've recorded a video demonstrating the entire process, including the test execution and the implementation details.

[Here](https://drive.google.com/file/d/1AwD8oDqrQDplqTj_hbrdEG0ncMzMujQO/view?usp=sharing) is a link to the video.

sharon-odhiambo reviewed (approved):
Hey @m-mukundi, great work on the Hooks, Support Commands and Scripts. Your code shows great implementation and the addition makes the code much cleaner and easily maintainable. You have also included detailed screenshots and videos which show how your code works and visuals of the automated tests running! Specifically like that you figured how to add the audio to your video!
Excellent work on this! Keep it up!
